### PR TITLE
Add support `RSpec/Rails/HttpStatus` when `have_http_status` with string argument

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-rspec/issues/1702

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/rspec/rails/http_status_spec.rb [hunk 1]
```diff
@@ -15,6 +15,17 @@
...
+    it 'registers an offense when using string value' do
+      expect_offense(<<-RUBY)
+        it { is_expected.to have_http_status "200" }
+                                             ^^^^^ Prefer `:ok` over `"200"` to describe HTTP status code.
+      RUBY
+
+      expect_correction(<<-RUBY)
+        it { is_expected.to have_http_status :ok }
+      RUBY
+    end
```
Comment (pirj) [inline spec/rubocop/cop/rspec/rails/http_status_spec.rb:27]: IDK if we should handle [this](https://github.com/rails/rails/blob/3668b4b5978822f1e6311c7fd7a32f58daee136a/actionpack/test/controller/render_test.rb#L950).

**Review by pirj — APPROVED** [review]
Thank you!

File: spec/rubocop/cop/rspec/rails/http_status_spec.rb (hunk 1 above)
Comment (ydah) [reply spec/rubocop/cop/rspec/rails/http_status_spec.rb:27]: Are you wondering whether to handle the following lines?

```ruby
    assert_equal 404, @response.response_code
    assert_equal "Not Found", @response.message
    assert_response :not_found
```

Comment (pirj) [reply spec/rubocop/cop/rspec/rails/http_status_spec.rb:27]: No, status code strings with additional data like status: "404 Eat Dirt" there.

File: lib/rubocop/cop/rspec/rails/http_status.rb [hunk 2]
```diff
@@ -195,6 +203,15 @@ def symbol
             def number
               node.source.to_i
             end
+
+            def nomalize_str
+              nomalized = node.source.delete('"')
+              if nomalized.match?(/\A\d+\z/)
+                ::Rack::Utils::SYMBOL_TO_STATUS_CODE.key(nomalized.to_i)
+              else
+                nomalized
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/rails/http_status.rb:212]: Typo:
```suggestion
            def normalize_str
              normalized = node.source.delete('"')
              if normalized.match?(/\A\d+\z/)
                ::Rack::Utils::SYMBOL_TO_STATUS_CODE.key(normalized.to_i)
              else
                normalized
```

Comment (ydah) [reply lib/rubocop/cop/rspec/rails/http_status.rb:212]: 😱 Thank you catching this!

**Review by bquorning — APPROVED** [review]

**Pull request merged**